x-pack/plugins/security_solution/public/common/components/events_viewer/event_details_flyout.tsx

```diff
@@ -34,35 +37,42 @@ const EventDetailsFlyoutComponent: React.FC<EventDetailsFlyoutProps> = ({
   browserFields,
   docValueFields,
   timelineId,
+  handleCloseExpandedEvent,
 }) => {
-  const dispatch = useDispatch();
   const expandedEvent = useDeepEqualSelector(
     (state) => state.timeline.timelineById[timelineId]?.expandedEvent ?? emptyExpandedEvent
   );
 
-  const handleClearSelection = useCallback(() => {
```

Comment:
maybe we could keep `handleClearSelection`, so we don't need to create multiple callbacks whenever we use `EventDetailsFlyout`?

Reply:
as in x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx
I pass `handleCloseExpandedEvent` to `EventsViewer` and `EventDetailsFlyout`, so I think it would be nice to define it in the same place and share it?

Reply:
it depends, in that case, I believe it will be cleaner to do it twice as we do have at least 3 flyouts that need this callback and 1 event details inside the timeline (and we will need them inside notes and pinned events as well), so with your solution, we will have to define this callback every time we want to use `EventDetails`, but the logic will be completely the same in all cases, so if it would be just single case then I'd agree that it's better to define them in one place and pass down, but as we want to use those components in multiple places, so I think it will be nicer to have it done inside the flyout and event details even if that means we have to create similar callback twice. We have just cleaned up the timeline from this pattern, so I just to make sure that we are defining callback where they are needed to avoid props drilling that makes code harder to read and maintain